decisions/wc_class_decisions.txt

```diff
@@ -1155,38 +1155,223 @@ decisions = {
+						this_would_see_prev_class_trait_if_prev_was_prevprev_class_trigger = yes
+					}
+				}
+			}
 		}
 		allow = {
-			show_only_failed_conditions = yes
+			# FROM = { NOT = { has_character_modifier = ask_to_be_mentor_cooldown } }
+		}
+		third_party_allow = {
+			# show_only_failed_conditions = yes
 			
-			ask_to_be_your_mentor_decision_allowed = yes
+			FROMFROM = {
+				show_scope_change = no
+				
+				ROOT = {
+					show_scope_change = no
+					
+					prev_student_this_class_mentor_allow = yes
+				}
+			}
 		}
 		effect = {
 			ROOT = {
 				save_event_target_as = mentor
 			}
-			FROM = {
+			FROMFROM = {
+				show_scope_change = no
+				
 				save_event_target_as = student
+
+				set_character_flag = ask_to_be_mentor_ongoing_flag
+
 				opinion = { modifier = opinion_teaching_me who = ROOT }
-				set_character_flag = sent_letter_to_mentor
+			}
+			FROM = {
+				save_event_target_as = target_initiator
+
+				# add_character_modifier = { name = ask_to_be_mentor_cooldown years = 1 }
+
 				character_event = { id = WCCLS.1500 }
 			}
 		}
+		ai_acceptance = {
+			factor = -20	# The base score
+			
+			# Opinion
+			# Positive
+			# ROOT likes FROM
+			additive_modifier = {
+				value = 20
+				localisation_key = ROOT_likes_FROM_tooltip
+				opinion = { who = FROM value >= 20 }
+				opinion = { who = FROM value < 40 }
+			}
+			additive_modifier = {
+				value = 40
+				localisation_key = ROOT_likes_FROM_tooltip
+				opinion = { who = FROM value >= 40 }
+				opinion = { who = FROM value < 60 }
+			}
+			additive_modifier = {
+				value = 60
+				localisation_key = ROOT_likes_FROM_tooltip
+				opinion = { who = FROM value >= 60 }
+				opinion = { who = FROM value < 80 }
+			}
+			additive_modifier = {
+				value = 80
+				localisation_key = ROOT_likes_FROM_tooltip
+				opinion = { who = FROM value >= 80 }
+				opinion = { who = FROM value < 100 }
+			}
+			additive_modifier = {
+				value = 100
+				localisation_key = ROOT_likes_FROM_tooltip
+				opinion = { who = FROM value == 100 }
+			}
+			# Negative
+			# ROOT dislikes FROM
+			additive_modifier = {
+				value = -20
+				localisation_key = ROOT_dislikes_FROM_tooltip
+				opinion = { who = FROM value <= -20 }
+				opinion = { who = FROM value > -40 }
+			}
+			additive_modifier = {
+				value = -40
+				localisation_key = ROOT_dislikes_FROM_tooltip
+				opinion = { who = FROM value <= -40 }
+				opinion = { who = FROM value > -60 }
+			}
+			additive_modifier = {
+				value = -60
+				localisation_key = ROOT_dislikes_FROM_tooltip
+				opinion = { who = FROM value <= -60 }
+				opinion = { who = FROM value > -80 }
+			}
+			additive_modifier = {
+				value = -80
+				localisation_key = ROOT_dislikes_FROM_tooltip
+				opinion = { who = FROM value <= -80 }
+				opinion = { who = FROM value > -100 }
+			}
+			additive_modifier = {
+				value = -100
+				localisation_key = ROOT_dislikes_FROM_tooltip
+				opinion = { who = FROM value == -100 }
+			}
```

Comment:
How about make it 10 per 10 instead of 20 per 20?
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46576860/70449993-0182b480-1ab4-11ea-8916-4449be255a0d.png)
![image](https://user-images.githubusercontent.com/46576860/70450025-0e070d00-1ab4-11ea-8912-4410c1e801e0.png)


</details>

Reply:
Added!
@zumbak04, check them, please